### PR TITLE
TensorFlow 入门指南

### LINKED ISSUE #20: Getting Started With TensorFlow
* Markdown文件：[文件地址](https://github.com/xitu/tensorflow/blob/zh-hans/tensorflow/docs_src/get_started/get_started.md)
  * PR 地址：#67
  ----
  * 翻译时间：7 天
  * 校对时间：2 天
  * 翻译积分：7 分
  * 校对积分：2+2 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by joyking7**
resolve #20

### CODE REVIEW

**Comment by joyking7** [comment]
@leviding  翻译完成

**Comment by shi-xiaopeng** [comment]
校对认领
@leviding

**Comment by leviding** [comment]
@shi-xiaopeng ok

**Comment by LilianYe** [comment]
校对认领
@leviding

**Comment by leviding** [comment]
@LilianYe 好的

File: tensorflow/docs_src/get_started/get_started.md [hunk 1]
```diff
@@ -383,67 +300,56 @@ train_input_fn = tf.estimator.inputs.numpy_input_fn(
...
-TensorFlow API.
+`tf.estimator` 并不会在它预定义的模型中限制你。假设我们想创建一个并没有在 TensorFlow 中创建的常见模型。我们仍然可以保留 `tf.estimator` 关于数据集、反馈、训练等高级的抽象。通过举例，我们将会展示如何利用我们关于低阶 TensorFlow API 知识来创建一个和 `LinearRegressor` 相同的模型。
 
-To define a custom model that works with `tf.estimator`, we need to use
-`tf.estimator.Estimator`. `tf.estimator.LinearRegressor` is actually
-a sub-class of `tf.estimator.Estimator`. Instead of sub-classing
-`Estimator`, we simply provide `Estimator` a function `model_fn` that tells
-`tf.estimator` how it can evaluate predictions, training steps, and
-loss. The code is as follows:
+创建一个使用 `tf.estimator` 的常见模型，我们需要使用 `tf.estimator.Estimator`。`tf.estimator.LinearRegressor` 实际上是 `tf.estimator.Estimator` 的子类。但是为了替代子类 `Estimator`，我们简单的通过一个 `model_fn` 函数来提供 `Estimator` ，从而来告诉 `tf.estimator` 如何进行评估预测、训练步骤和损耗计算。代码如下：
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: ["但是为了替代子类 `Estimator`，我们简单的通过一个 `model_fn` 函数来提供 `Estimator` ，从而来告诉 `tf.estimator` 如何进行评估预测、训练步骤和损耗计算"]
=>["相比于实现一个`Estimator`的子类，我们只简单地通过给 `Estimator` 提供一个 `model_fn` 函数来告诉 `tf.estimator`如何进行预测、训练和损耗函数的计算。"]
看来译者也是尽力了，但是没有实际经验的话，确实无法理解这其中的意思。sub_classing的意思是子类化，即创建一个类集成“Estimator”,然后在新建类中写控制方法；文中的意思是，直接在父类中写这个控制方法，model_fn, 省去了创建子类，所以说“简单地”。

File: tensorflow/docs_src/get_started/get_started.md [hunk 2]
```diff
@@ -383,67 +300,56 @@ train_input_fn = tf.estimator.inputs.numpy_input_fn(
...
 
 ```python
 import numpy as np
 import tensorflow as tf
 
-# Declare list of features, we only have one real-valued feature
+# 声明一系列特征，这里我们只有实值特征
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: 译为 “声明特征列表” 更好，这里的意思不是声明一系列特征，而是声明数据的全部特征。

File: tensorflow/docs_src/get_started/get_started.md [hunk 3]
```diff
@@ -383,67 +300,56 @@ train_input_fn = tf.estimator.inputs.numpy_input_fn(
...
   loss = tf.reduce_sum(tf.square(y - labels))
-  # Training sub-graph
+  # 训练的子图
   global_step = tf.train.get_global_step()
   optimizer = tf.train.GradientDescentOptimizer(0.01)
   train = tf.group(optimizer.minimize(loss),
                    tf.assign_add(global_step, 1))
-  # EstimatorSpec connects subgraphs we built to the
-  # appropriate functionality.
+  # EstimatorSpec 连接了我们所创建的相关函数的子图
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: [“EstimatorSpec 连接了我们所创建的相关函数的子图”]
=> 应该译为 [“ EstimatorSpec 连接我们所创建的子图到相关的函数”]
意思错了，主体是：" ... connects subgraphs to ..."

File: tensorflow/docs_src/get_started/get_started.md [hunk 4]
```diff
@@ -455,26 +361,22 @@ train_input_fn = tf.estimator.inputs.numpy_input_fn(
...
-When run, it produces
+运行之后，结果如下：
 ```
 train metrics: {'loss': 1.227995e-11, 'global_step': 1000}
 eval metrics: {'loss': 0.01010036, 'global_step': 1000}
 ```
 
-Notice how the contents of the custom `model_fn()` function are very similar
-to our manual model training loop from the lower level API.
+可以注意到，自定义 `model_fn()` 函数的结果和我们利用低阶 API 进行手动训练循环的结果十分相似。
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: 不是结果相似，而是内容相似！
应译为:  “可以注意到，这个自定义 `model_fn()` 函数的代码和我们利用低阶 API 进行手动训练的模型（model）代码是多么地相似”

File: tensorflow/docs_src/get_started/get_started.md [hunk 5]
```diff
@@ -1,107 +1,75 @@
...
+# 张量
+
+TensorFlow 的核心数据单位就是**张量**。张量可看做是由原始值组成的 n 维度数组。张量的**阶**就是它的维度。下面是一些关于张量的例子：
 
 ```python
-3 # a rank 0 tensor; a scalar with shape []
-[1., 2., 3.] # a rank 1 tensor; a vector with shape [3]
-[[1., 2., 3.], [4., 5., 6.]] # a rank 2 tensor; a matrix with shape [2, 3]
-[[[1., 2., 3.]], [[7., 8., 9.]]] # a rank 3 tensor with shape [2, 1, 3]
+3 # 0 阶张量，纯量（只有大小）
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: shape 是数据的一个重要特征，不可以缺少。
应译为 “0 阶张量，标量的 shape 为 [ ]”

File: tensorflow/docs_src/get_started/get_started.md [hunk 6]
```diff
@@ -1,107 +1,75 @@
...
+[1., 2., 3.] # 1 阶张量，向量（大小和方向）
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: ["1 阶张量，向量（大小和方向）"]
=> [" 1 阶张量，此向量的shape为 [ 3 ] "]

File: tensorflow/docs_src/get_started/get_started.md [hunk 7]
```diff
@@ -1,107 +1,75 @@
...
+[[1., 2., 3.], [4., 5., 6.]] # 2 阶张量，矩阵（数据表）
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: ["2 阶张量，矩阵（数据表）"]
=> [" 2 阶张量，此矩阵的shape为[2, 3] "]

File: tensorflow/docs_src/get_started/get_started.md [hunk 8]
```diff
@@ -1,107 +1,75 @@
...
+[[[1., 2., 3.]], [[7., 8., 9.]]] # 3 阶张量，数据立体
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: ["3 阶张量，数据立体"]
=> ["3 阶张量，shape 为 [2, 1, 3] "]

File: tensorflow/docs_src/get_started/get_started.md [hunk 9]
```diff
@@ -1,107 +1,75 @@
...
-tensors:
+# TensorFlow 入门指南
+
+本指南可以让你在 TensorFlow 中进行编程。在开始本指南之前，请先[安装 TensorFlow](https://www.tensorflow.org/install/index)。为了能够更好的利用本指南，你应当了解如下知识：
+
+*   如何使用 Python 编程
+*   至少了解数组知识
+*   理想情况下了解一些机器学习知识，当然，即使你不了解机器学习，本指南也会是你应该阅读的第一个指南。
+
+TensorFlow 提供了多种 API。其中最底层的 API —— TensorFlow Core 能够为你提供完整的编程控制功能。我们向机器学习研究人员和其他需要对模型进行良好控制人员推荐使用 TensorFlow Core。更高级的 API 是构建在 TensorFlow Core 之上。这些更高级的 API 通常比 TensorFlow Core 更容易学习和使用。此外，这些更高级的 API 使得不同用户之间的重复性任务更加简单且更具有一致性。像 `tf.setimator` 这样高级的 API 能够帮你管理数据集、评估器、训练以及推理。
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: 最后一句 “estimator“ 拼错了！

File: tensorflow/docs_src/get_started/get_started.md [hunk 10]
```diff
@@ -1,107 +1,75 @@
...
 
 ```python
 import tensorflow as tf
 ```
-This gives Python access to all of TensorFlow's classes, methods, and symbols.
-Most of the documentation assumes you have already done this.
+这使 Python 可以使用 TensorFlow 所有的类、方法和符号。绝大部分文档都会假定你已经完成了导入的部分。
 
-### The Computational Graph
+### 构建数据流图（或计算图）
```
Comment (shi-xiaopeng) [inline tensorflow/docs_src/get_started/get_started.md]: 统一用”计算图“来表示 computational graph

**Comment by shi-xiaopeng** [comment]
@leviding 校对完成啦！！别忘记你的诺言哦 :D

**Comment by leviding** [comment]
2333 @shi-xiaopeng 翻倍

**Comment by joyking7** [comment]
@shi-xiaopeng 已经根据意见进行修改，自己当时理解也不是很清楚，通过校对的意见，对于自己当时不懂得地方也理解了，感谢校对~

File: tensorflow/docs_src/get_started/get_started.md [hunk 11]
```diff
@@ -185,193 +139,156 @@ x = tf.placeholder(tf.float32)
...
-The final print shows the loss now is zero.
+最终的输出显示现在的损耗为零:
 ```
 0.0
 ```
 
-We guessed the "perfect" values of `W` and `b`, but the whole point of machine
-learning is to find the correct model parameters automatically.  We will show
-how to accomplish this in the next section.
+我们猜测了“最优”的 `W` 和 `b` 值，但是机器学习的整个重点是自动寻找正确的模型参数。我们将会在下一节中展示如何完成这项工作。
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 机器学习的整个重点-> 整个机器学习的重点

File: tensorflow/docs_src/get_started/get_started.md [hunk 12]
```diff
@@ -185,193 +139,156 @@ x = tf.placeholder(tf.float32)
...
-A complete discussion of machine learning is out of the scope of this tutorial.
-However, TensorFlow provides **optimizers** that slowly change each variable in
-order to minimize the loss function. The simplest optimizer is **gradient
-descent**. It modifies each variable according to the magnitude of the
-derivative of loss with respect to that variable. In general, computing symbolic
-derivatives manually is tedious and error-prone. Consequently, TensorFlow can
-automatically produce derivatives given only a description of the model using
-the function `tf.gradients`. For simplicity, optimizers typically do this
-for you. For example,
+对于机器学习完整的讨论已经超出了本教程的范围。然而，TensorFlow 提供了**优化器**来缓慢地更改每个变量，从而最大程度的降低损耗函数。最简单的优化器就是**梯度下降**。它根据相对于变量的损耗函数导数的大小来修改每个变量。通常来说，手动计算函数导数是很乏味且易出错的。因此，TensorFlow 可以使用 `tf.gradients` 来自动的生成仅给出了模型描述的导数。为了方便起见，优化器通常会帮助用户做这样的操作。例如，
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 因此，TensorFlow 可以使用 `tf.gradients` 来自动的生成仅给出了模型描述的导数。
TensorFlow 可以只根据模型描述用 `tf.gradients` 自动的生成导数。

File: tensorflow/docs_src/get_started/get_started.md [hunk 13]
```diff
@@ -185,193 +139,156 @@ x = tf.placeholder(tf.float32)
...
 
 ```python
 optimizer = tf.train.GradientDescentOptimizer(0.01)
 train = optimizer.minimize(loss)
 ```
 
 ```python
-sess.run(init) # reset values to incorrect defaults.
+sess.run(init) # 将数值重置为不准确的默认值
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 不准确->不正确

File: tensorflow/docs_src/get_started/get_started.md [hunk 14]
```diff
@@ -185,193 +139,156 @@ x = tf.placeholder(tf.float32)
...
 ```python
-# NumPy is often used to load, manipulate and preprocess data.
+# NumPy 经常被用来载入、操纵和预处理数据集
 import numpy as np
 import tensorflow as tf
 
-# Declare list of features. We only have one numeric feature. There are many
-# other types of columns that are more complicated and useful.
+# 声明一系列特征。这里我们只有一组纯数字特征
+# 还有许多其他类型的更复杂更有用的 column
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 声明特征列表。
tf.feature_column中还有许多其他类型的更复杂更有用的 column

File: tensorflow/docs_src/get_started/get_started.md [hunk 15]
```diff
@@ -185,193 +139,156 @@ x = tf.placeholder(tf.float32)
...
 feature_columns = [tf.feature_column.numeric_column("x", shape=[1])]
 
-# An estimator is the front end to invoke training (fitting) and evaluation
-# (inference). There are many predefined types like linear regression,
-# linear classification, and many neural network classifiers and regressors.
-# The following code provides an estimator that does linear regression.
+# 估算器可以调用训练（拟合）和评估（推理）方法的前端
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 估算器是

File: tensorflow/docs_src/get_started/get_started.md [hunk 16]
```diff
@@ -383,67 +300,56 @@ train_input_fn = tf.estimator.inputs.numpy_input_fn(
...
 ```
-When run, it produces something like
+运行之后，产生如下结果：
 ```
 train metrics: {'average_loss': 1.4833182e-08, 'global_step': 1000, 'loss': 5.9332727e-08}
 eval metrics: {'average_loss': 0.0025353201, 'global_step': 1000, 'loss': 0.01014128}
 ```
-Notice how our eval data has a higher loss, but it is still close to zero.
-That means we are learning properly.
+注意到我们的评估数据集有更高的损耗值，但是他仍然接近于零。这就说明我们的学习比较适合。
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 注意到我们在评估数据集上有更高的损耗值，但是它仍然接近于零。

File: tensorflow/docs_src/get_started/get_started.md [hunk 17]
```diff
@@ -455,26 +361,22 @@ train_input_fn = tf.estimator.inputs.numpy_input_fn(
 eval_input_fn = tf.estimator.inputs.numpy_input_fn(
     {"x": x_eval}, y_eval, batch_size=4, num_epochs=1000, shuffle=False)
 
-# train
+# 训练
 estimator.train(input_fn=input_fn, steps=1000)
-# Here we evaluate how well our model did.
+# 这里对于我们模型的性能进行评估
```
Comment (LilianYe) [inline tensorflow/docs_src/get_started/get_started.md]: 对于->对

**Comment by LilianYe** [comment]
@joyking7 @leviding 校对完成。第一次校对，有做的不对的地方还请指正。

**Comment by leviding** [comment]
@LilianYe 很好

**Comment by joyking7** [comment]
@shi-xiaopeng @LilianYe @leviding 已经根据两位校对者的意见进行了修改，辛苦校对者的校对。

**Comment by leviding** [comment]
@joyking7 更新及时，加 1 分。
